app/models/subscription.rb

```diff
@@ -2,19 +2,38 @@ class Subscription < ApplicationRecord
   belongs_to :user
   belongs_to :plan
   has_many :transactions
+  has_many :usages, dependent: :destroy
 
   validates :billing_day, presence: true, inclusion: 1..28
 
   before_validation :set_billing_day
   after_save :subscription_transaction
 
+  def charge_fee
+    subscription_transaction
+  end
```

Comment:
how will you check if this `charge_fee` is being called on the billing day?

Reply:
On view level we do not show admin `charge` option.

On Model level we can implement a condition like this:

```
  def charge_fee
    is_billing_day?(billing_day) ? subscription_transaction : false
  end
```

Reply:
yes, something like this. But you can make this more simpler.